app/components/AppUpdateNotification.tsx

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable react/no-unused-prop-types */
+import React, {
+    useState,
+    useCallback,
+    useEffect,
+    useMemo,
+    Dispatch,
+    SetStateAction,
+    ComponentType,
+} from 'react';
+
+import CogIcon from '@resources/svg/settings.svg';
+import ErrorIcon from '@resources/svg/logo-error.svg';
+import CheckmarkIcon from '@resources/svg/logo-checkmark.svg';
+import Button from '~/cross-app-components/Button';
+import { NotificationLevel } from '~/features/NotificationSlice';
+import Loading from '~/cross-app-components/Loading';
+
+import Notification from './Notification';
+import Countdown from './Countdown';
+
+enum UpdateStatus {
+    Available,
+    Downloading,
+    Verifying,
+    Success,
+    Error,
+}
+interface Props {
+    version: string;
+    onUpdate(): void;
+    onClose(): void;
+}
+
+interface UpdateState {
+    status: UpdateStatus;
+    message?: string;
+}
+
+interface ChildProps extends Props, UpdateState {
+    setState: Dispatch<SetStateAction<UpdateState>>;
+}
+
+const AvailableChild = ({
+    onUpdate,
+    setState,
+    onClose,
+    version,
+}: ChildProps) => {
+    const update = useCallback(() => {
+        onUpdate();
+        setState({ status: UpdateStatus.Downloading });
+    }, [onUpdate, setState]);
+
+    return (
+        <>
+            <div className="flex alignCenter">
+                <CogIcon width="21" height="22" />
+                <span className="mL5">Version {version} is available.</span>
+            </div>
+            <div className="inlineFlexColumn mT20">
+                <Button size="tiny" onClick={update}>
+                    Restart & install
+                </Button>
+                <Button className="mT10" size="tiny" onClick={onClose}>
+                    Remind me
+                </Button>
+            </div>
+        </>
+    );
+};
+
+const iconWidth = 60;
+
+const ProcessChild = ({ status }: ChildProps) => (
+    <>
+        <Loading inline iconWidth={iconWidth} />
+        <div className="mT10 textCenter">
+            {status === UpdateStatus.Downloading && 'Downloading update'}
```

Comment:
While testing I tried to mess up some of the naming/versioning so they mismatched. In that case it cannot find the update:
```
Error: Error: Cannot download "https://github.com/orhoj/concordium-desktop-wallet/releases/download/v1.2.0-alpha2/concordium-desktop-wallet-1.2.0-alpha2.AppImage", status 404:
```
and the UI keeps saying `Downloading update`. It should update the status in this case instead.

Reply:
good catch, added error handling which should fix this.